src/cpu/kvm/perfevent.cc

```diff
@@ -67,33 +68,54 @@ PerfKvmCounterConfig::~PerfKvmCounterConfig()
 {
 }
 
-
-PerfKvmCounter::PerfKvmCounter(PerfKvmCounterConfig &config, pid_t tid)
-    : fd(-1), ringBuffer(NULL), pageSize(-1)
+PerfKvmCounter::PerfKvmCounter()
 {
-    attach(config, tid, -1);
 }
 
-PerfKvmCounter::PerfKvmCounter(PerfKvmCounterConfig &config,
-                         pid_t tid, const PerfKvmCounter &parent)
-    : fd(-1), ringBuffer(NULL), pageSize(-1)
+PerfKvmCounter *
+PerfKvmCounter::create(bool allow_hybrid)
 {
-    attach(config, tid, parent);
+    // Check if we're running on a hybrid host architecture. Linux exposes
+    // this via sysfs. If the directory /sys/devices/cpu exists, then we are
+    // running on a regular architecture. Otherwise, /sys/devices/cpu_core and
+    // /sys/devices/cpu_atom should exist. For simplicity, we use the
+    // existence of /sys/devices/cpu_atom to indicate a hybrid host
+    // architecture.
+    const char *atom_path = "/sys/devices/cpu_atom";
```

Comment:
It'd be nice if we could use std::filesystem instead but that requires C++17, so it might not be an option yet.